src/options.ml

```diff
@@ -412,13 +412,22 @@ let init () =
          | Some sep -> Some (String.before config sep, String.after config (sep + 2))
          | None -> None)
       ocamlc_config_lines in
+  let get_field x =
+    match List.assoc_opt (x : string) ocamlc_configs with
+    | Some s -> s
+    | None -> failwith (x^" could not be found in ocamlc -config")
+  in
   let get_ext x =
-    let s = List.assoc x ocamlc_configs in
-    if String.length s > 0 then String.after s 1 else s in
+    let s = get_field x in
+    if String.length s > 0 && s.[0] = '.' then
+      String.after s 1
+    else
+      s
+  in
   ext_lib := get_ext "ext_lib";
   ext_obj := get_ext "ext_obj";
   ext_dll := get_ext "ext_dll";
-  exe := List.assoc "ext_exe" ocamlc_configs;
+  exe := get_field "ext_exe";
```

Comment:
Thank you very much for taking over!

That line was not a mistake though (actually I wrote what you wrote at first too and got caught up by the test suite under windows). `exe` is supposed to keep its initial dot while others don't! (Yes I refrained myself to uninformize that as I tried to keep the change minimal but do what you want, you're a maintainer :-) )

Reply:
it does keep its initial dot though. `get_field` and `get_ext` are two distinct functions

Reply:
btw did the testsuite work under windows for you? The CI also failed with your commit on Windows

Reply:
My bad!
(I retry on windows right now)

Reply:
i've renamed the two functions and gave them different roles to avoid similar confusion in the future.